oz/sqlalchemy/middleware.py

```diff
@@ -26,9 +31,26 @@ def _sqlalchemy_on_finish(self):
         """
 
         if hasattr(self, "db_conn"):
-            if self.get_status() >= 200 and self.get_status() <= 399:
-                self.db_conn.commit()
-            else:
-                self.db_conn.rollback()
+            try:
+                if self.get_status() >= 200 and self.get_status() <= 399:
+                    self.db_conn.commit()
+                else:
+                    self.db_conn.rollback()
+            except:
+                tornado.log.app_log.warning("Error occurred during database transaction cleanup: %s", str(sys.exc_info()[0]))
```

Comment:
To an extent, but I'm not just worried about sqlalchemy, since this is an unconditional `except` (one of the few times I actually want to use an unconditional except). We could use a boolean setting called `INCLUDE_TRACEBACKS` and adjust how we log based on it.

I don't think we need to accept that the logs will get polluted with sensitive data, especially not for an open-source library where we cannot predict the context of its use. I reraise the exception exactly for this reason: if someone wants to further inspect the exception, they can catch it themselves. I think any library should assume the least flexible environment when possible.